riscv/mmu.h

```diff
@@ -161,6 +167,12 @@ class mmu_t
     store(addr, val, {forced_virt, hlvx, lr});
   }
 
+  // shadow stack store
+  template<typename T>
+  void ss_store(reg_t addr, T val) {
+    store<T>(addr, val, {false, false, false, /*shstk=*/ true});
```

Comment:
This `xlate_flags_t` is getting unwieldy. If we could upgrade to use C++20, we could use [designated initializers](https://www.cppstories.com/2021/designated-init-cpp20/) and make this much more readable.

cc @rbuchner-aril

Reply:
Would upgrading to C++20 be accepted as would force users to upgrade toolchains? If so, I can put our change to using designated initializers into a PR.

Reply:
I'm generally hesitant to force people to upgrade compiler versions to address our minor inconveniences, but in this case, designated initializer support has been present in GCC since version 8, which rolled out just under 6 years ago.  Clang support came a bit later (v10, just under 4 years ago).

I think we should not update the `-std` command-line arg to `c++20`, though, since that needlessly forces an even more recent compiler; and, in any case, Clang and GCC both appear to support designated initializers without that flag.

OK with you, too, @jerryz123?

Reply:
This seems reasonable.

Reply:
@aswaterman Without updating to `-std=c++2a` (leaving Makefile with `-std=c++17`, I see this warning after adding the designated-initializers changes.

```
/local_home/rbuchner/spike-regress/riscv-isa-sim/riscv/mmu.h:42:30: warning: default member initializers for bit-fields only available with -std=c++2a or -std=gnu++2a
   const bool forced_virt : 1 {false};
```

We are running GCC 8.2.0:
```console
$ which gcc
/tools/gnu/gcc/8.2.0/bin/gcc
```

FWIW, the compile still succeeds and passed our tests despite this warning, but this isn't safe.

Could implement with just `designated-initializers` and not the default member initializers but that doesn't seem to be much of an improvement over using `/*field_name*/` to label the structures.

Reply:
#1583 is now merged, so this PR will need some work to rebase onto that, but it will really clean this up.

Reply:
I already fixed this, thanks.